system/src/Grav/Common/Uri.php

```diff
@@ -157,14 +157,26 @@ private function buildHost()
         return $this->name;
     }
 
+    /**
+     * Return a string with the current environment name.
+     *
+     * If the server IP is a local server address (127.0.0.1 or ::1), and/or
+     * the host name is set to 'localhost', returns 'localhost'.
+     *
+     * Otherwise it returns the server name as set by self::buildHostname()
+     *
+     * @return string
+     */
     private function buildEnvironment()
     {
-        // set hostname
-        $address = isset($_SERVER['REMOTE_ADDR']) ? $_SERVER['REMOTE_ADDR'] : '::1';
+        $address = isset($_SERVER['SERVER_ADDR']) ? $_SERVER['SERVER_ADDR'] : '::1';
 
-        // check for localhost variations
         if ($this->name == 'localhost' || $address == '::1' || $address == '127.0.0.1') {
-            $env = 'localhost';
+            if ($this->name != 'localhost') {
+                $env = $this->name;
+            } else {
+                $env = 'localhost';
+            }
         } else {
             $env = $this->name;
         }
```

Comment:
We follow/use git-flow.

`master` is the latest release (stable) branch. 
`develop` is the working branch. PRs are automatically sent to the `develop` branch, so no worries.

This is for Grav core and all plugins/themes.

You can find some more info on https://github.com/getgrav/grav/blob/develop/CONTRIBUTING.md